website/server/controllers/api-v3/auth.js

```diff
@@ -584,8 +608,14 @@ api.updateEmail = {
 
     if (emailAlreadyInUse) throw new NotAuthorized(res.t('cannotFulfillReq'));
 
-    let candidatePassword = passwordUtils.encrypt(req.body.password, user.auth.local.salt);
-    if (candidatePassword !== user.auth.local.hashed_password) throw new NotAuthorized(res.t('wrongPassword'));
+    let password = req.body.password;
```

Comment:
I'm not sure I agree with this:

1. `Throwing the error in the compare function`:  we may want to show a different error in some situations and right now there's a piece of code where such a change would result in a duplication of the code that throws the error

```
    let isValidPassword;

    if (!user) {
      isValidPassword = false;
    } else {
      isValidPassword = await passwordUtils.compare(user, password);
    }

    if (!isValidPassword) throw new NotAuthorized(res.t('invalidLoginCredentialsLong'));
```

2. Converting to bcrypt while comparing: I would prefer not because we're talking about two very different actions: one is checking for the validity of the password and the other is converting data to something different (and requires a save afterwards). And for example when deleting an user account we don't care about converting the password to bcrypt before it's destroyed